web/src/engine/websites/CyComi.ts

```diff
@@ -24,17 +71,17 @@ export default class extends DecoratableMangaScraper {
     public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
         const id = new URL(url).pathname.split('/').pop();
         const request = new Request(`${this.apiURL}/title/detail?titleId=${id}`);
-        const { data } = await FetchJSON<APIManga>(request);
+        const { data } = await FetchJSON<APIResult<APIManga>>(request);
         return new Manga(this, provider, data.titleId.toString(), data.titleName);
     }
 
     public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
         const mangaList: Manga[] = [];
-        for(let page = 0, run = true; run; page++) {
+        for (let page = 0, run = true; run; page++) {
             const request = new Request(`${this.apiURL}/home/paginatedList?limit=${50}&page=${page}`);
-            const { data } = await FetchJSON<APIMangas>(request);
-            const mangas = data.reduce((accumulator: Manga[], entry) => {
-                if(entry) {
+            const { data, resultCode } = await FetchJSON<APIResult<APIMangas[]>>(request);
+            const mangas = resultCode !== 1 || !data ? [] : data.reduce((accumulator: Manga[], entry) => {
+                if (entry) {

```

Comment:
Based on the type definition `entry` cannot be null.
Either make `APIMangas` nullable, or remove this conditional check.

Reply:
Hum, yeah.

![image](https://github.com/manga-download/haruneko/assets/10671518/57d0a54f-f7b0-41b8-aeff-d8aac75b6153)